lib/Service/Group/NextcloudGroupService.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * @author Matthias Rella <mrella@pisys.eu>
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Service\Group;
+
+use OCP\IGroupManager;
+
+class NextcloudGroupService extends AbstractGroupService {
+
+	/**
+	 * Nextcloud's group manager
+	 *
+	 * @var IGroupManager
+	 */
+	private $groupManager;
+
+	/**
+	 * Group's namespace
+	 *
+	 * @var string
+	 */
+	private $namespace = "Nextcloud";
+
+	public function __construct(IGroupManager $groupManager) {
+		$this->groupManager = $groupManager;
+	}
+
+	public function getNamespace() {
+		return $this->namespace;
+	}
+
+	public function search($term) {
+		$groups = $this->groupManager->search($term);
+
+		return array_map(
+			function($g) {
+				return [
```

Comment:
Looks like a good candidate to implement PHP's `JsonSerializable` interface for this.

Reply:
I ported this pattern from `Contacts\IManager::search`. It's just returning key/value pairs. Where would `JsonSerializable` fit in?

Reply:
Nvm, just realized that `$g` is some data structure from the Nextcloud server, so nothing we can change. In that case that's fine :+1: